Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs

```diff
@@ -118,36 +138,33 @@ internal sealed class IntellisenseController : IIntellisenseController, IOleComm
         }
 
         public void ConnectSubjectBuffer(ITextBuffer subjectBuffer) {
-            PropagateAnalyzer(subjectBuffer);
+            _subjectBuffers.Add(subjectBuffer);
 
-            Debug.Assert(_bufferParser != null, "SetBufferParser has not been called");
-            if (_bufferParser == null) {
-                return;
+            var analyzer = _textView.GetAnalysisEntry(subjectBuffer, _serviceProvider)?.Analyzer;
+            bool isTemporaryFile = false;
+            if (analyzer == null) {
+                // there's no analyzer for this file, but we can analyze it against either
+                // the default analyzer or some other analyzer (e.g. if it's a diff view, we want
+                // to analyze against the project we're diffing from).  But in either case this
+                // is just a temporary file which should be closed when the view is closed.
+                analyzer = _textView.GetBestAnalyzer(_serviceProvider);
+                isTemporaryFile = true;
             }
 
-            BufferParser existingParser;
-            if (!subjectBuffer.Properties.TryGetProperty(typeof(BufferParser), out existingParser)) {
-                _bufferParser.AddBuffer(subjectBuffer);
-            } else {
-                // already connected to a buffer parser, we should have the same project entry
-                Debug.Assert(_bufferParser._analysis == existingParser._analysis);
+            if (analyzer != null) {
+                analyzer.MonitorTextBufferAsync(subjectBuffer, isTemporaryFile).ContinueWith(task => {
```

Comment:
`ContinueWith` does not marshal back to the original thread by default, but we need to if we want `AttachedViews++` to be safe.

Probably more reliable just to lock the parser while incrementing though.